RC4/5: NPE in FilesSyncHelper.scheduleNJobs

```
java.lang.RuntimeException: 
  at android.app.job.JobService$JobHandler.handleMessage (JobService.java:147)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6688)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1468)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1358)
Caused by: java.lang.NullPointerException: 
  at com.owncloud.android.utils.FilesSyncHelper.scheduleNJobs (FilesSyncHelper.java:258)
  at com.owncloud.android.jobs.NContentObserverJob.onStartJob (NContentObserverJob.java:59)
  at android.app.job.JobService$JobHandler.handleMessage (JobService.java:143)
```

LoC:
```
SyncedFolderProvider syncedFolderProvider = new 
SyncedFolderProvider(MainApp.getAppContext().getContentResolver());
```
I'd say ```MainApp.getAppContext()``` in ```null```

--- a/src/main/java/com/owncloud/android/MainApp.java
+++ b/src/main/java/com/owncloud/android/MainApp.java
@@ -186,7 +186,7 @@ public class MainApp extends MultiDexApplication {
 
         initiateExistingAutoUploadEntries();
 
-        FilesSyncHelper.scheduleFilesSyncIfNeeded();
+        FilesSyncHelper.scheduleFilesSyncIfNeeded(mContext);
         FilesSyncHelper.restartJobsIfNeeded();
 
         ReceiversHelper.registerNetworkChangeReceiver();

--- a/src/main/java/com/owncloud/android/jobs/NContentObserverJob.java
+++ b/src/main/java/com/owncloud/android/jobs/NContentObserverJob.java
@@ -56,7 +56,7 @@ public class NContentObserverJob extends JobService {
                         .schedule();
             }
 
-            FilesSyncHelper.scheduleNJobs(true);
+            FilesSyncHelper.scheduleNJobs(true, getApplicationContext());
         }
 
         return true;

--- a/src/main/java/com/owncloud/android/ui/activity/SyncedFoldersActivity.java
+++ b/src/main/java/com/owncloud/android/ui/activity/SyncedFoldersActivity.java
@@ -502,7 +502,7 @@ public class SyncedFoldersActivity extends FileActivity implements SyncedFolderA
             String syncedFolderInitiatedKey = "syncedFolderIntitiated_" + syncedFolderDisplayItem.getId();
             arbitraryDataProvider.deleteKeyForAccount("global", syncedFolderInitiatedKey);
         }
-        FilesSyncHelper.scheduleNJobs(false);
+        FilesSyncHelper.scheduleNJobs(false, getApplicationContext());
 
     }
 
@@ -555,7 +555,7 @@ public class SyncedFoldersActivity extends FileActivity implements SyncedFolderA
                     String syncedFolderInitiatedKey = "syncedFolderIntitiated_" + newCustomFolder.getId();
                     arbitraryDataProvider.deleteKeyForAccount("global", syncedFolderInitiatedKey);
                 }
-                FilesSyncHelper.scheduleNJobs(false);
+                FilesSyncHelper.scheduleNJobs(false, getApplicationContext());
             }
             mAdapter.addSyncFolderItem(newCustomFolder);
         } else {
@@ -575,7 +575,7 @@ public class SyncedFoldersActivity extends FileActivity implements SyncedFolderA
                         String syncedFolderInitiatedKey = "syncedFolderIntitiated_" + item.getId();
                         arbitraryDataProvider.deleteKeyForAccount("global", syncedFolderInitiatedKey);
                     }
-                    FilesSyncHelper.scheduleNJobs(false);
+                    FilesSyncHelper.scheduleNJobs(false, getApplicationContext());
                 }
             } else {
                 // existing synced folder setup to be updated
@@ -586,7 +586,7 @@ public class SyncedFoldersActivity extends FileActivity implements SyncedFolderA
                     String syncedFolderInitiatedKey = "syncedFolderIntitiated_" + item.getId();
                     arbitraryDataProvider.deleteKeyForAccount("global", syncedFolderInitiatedKey);
                 }
-                FilesSyncHelper.scheduleNJobs(false);
+                FilesSyncHelper.scheduleNJobs(false, getApplicationContext());
             }
 
             mAdapter.setSyncFolderItem(syncedFolder.getSection(), item);

--- a/src/main/java/com/owncloud/android/utils/FilesSyncHelper.java
+++ b/src/main/java/com/owncloud/android/utils/FilesSyncHelper.java
@@ -253,9 +253,8 @@ public class FilesSyncHelper {
         return false;
     }
 
-    public static void scheduleNJobs(boolean force) {
-        SyncedFolderProvider syncedFolderProvider = new SyncedFolderProvider(MainApp.getAppContext().
-                getContentResolver());
+    public static void scheduleNJobs(boolean force, Context context) {
+        SyncedFolderProvider syncedFolderProvider = new SyncedFolderProvider(context.getContentResolver());
 
 
         boolean hasVideoFolders = false;
@@ -282,7 +281,7 @@ public class FilesSyncHelper {
             }
     }
 
-    public static void scheduleFilesSyncIfNeeded() {
+    public static void scheduleFilesSyncIfNeeded(Context context) {
         // always run this because it also allows us to perform retries of manual uploads
         new JobRequest.Builder(FilesSyncJob.TAG)
                 .setPeriodic(900000L, 300000L)
@@ -290,7 +289,9 @@ public class FilesSyncHelper {
                 .build()
                 .schedule();
 
-        scheduleNJobs(false);
+        if (context != null) {
+            scheduleNJobs(false, context);
+        }
     }
 
     @RequiresApi(api = Build.VERSION_CODES.N)

